webapp/src/main/java/com/box/l10n/mojito/service/branch/BranchService.java

```diff
@@ -87,4 +97,28 @@ public PollableFuture<Void> asyncDeleteBranch(Long repositoryId, Long branchId)
             .build();
     return quartzPollableTaskScheduler.scheduleJob(quartzJobInfo);
   }
+
+  public void addBranchSource(Branch branch) {
+    // Mojito push links text unit extractions to empty branch, don't attempt to update the source
+    if (branch.getName() == null) return;
+
+    com.box.l10n.mojito.service.branch.BranchSource branchSource =
+        branchSourceConfig.getBranchSources().get(branch.getRepository().getName());
```

Comment:
Does this mean we need config in `app.properties` for each individual repository?

Reply:
Yes, you may want a repository to have a different source link / may need to customize it more for that repository. Any other ideas come to mind that wouldn't require us to have a definition for all repositories ? I could introduce a app property e.g `l10n.branchSources.*.url=https://github.com/{branchName}`. All lookups will attempt to find the repository config and if that fails it could use the `*` wildcard.

Reply:
Also just realised that I can also attach it to the `l10n.link.repositoryname` app property.

Reply:
Done @maallen